src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -1738,6 +1751,69 @@ public static void ConsumeLotsOfComments(string valueString, bool insideArray, s
             Assert.Equal(dataUtf8.Length, json.BytesConsumed);
         }
 
+        [Theory]
+        [MemberData(nameof(SingleLineCommentData))]
+        public static void ConsumeSingleLineCommentSingleSpanTest(string expected, string[] inputData)
```

Comment:
These tests could also be written by looping through and splitting the input at every index. Something like this:
```C#
[Theory]
[MemberData(nameof(SingleLineCommentData))]
public static void ConsumeSingleLineCommentSingleSpanTest(string inputData, string expected)
{
    byte[] dataUtf8 = Encoding.UTF8.GetBytes(inputData);
    for (int i = 0; i < inputData.Length; i++)
    {
        var state = new JsonReaderState(options: new JsonReaderOptions { CommentHandling = JsonCommentHandling.Allow });
        var json = new Utf8JsonReader(dataUtf8.AsSpan(i), isFinalBlock: false, state);
        VerifyReadLoop(ref json, expected);

        json = new Utf8JsonReader(dataUtf8.AsSpan((int)state.BytesConsumed), isFinalBlock: true, state);
        VerifyReadLoop(ref json, expected);
    }
}

private static void VerifyReadLoop(ref Utf8JsonReader json, string expected)
{
    while (json.Read())
    {
        switch (json.TokenType)
        {
            case JsonTokenType.StartObject:
            case JsonTokenType.EndObject:
                break;
            case JsonTokenType.Comment:
                Assert.Equal(Encoding.UTF8.GetBytes(expected), json.ValueSpan.ToArray());
                break;
            default:
                Assert.True(false);
                break;
        }
    }
}
```

Reply:
In fact, we have some existing tests that follow this pattern of splitting the input in two parts at every location:
https://github.com/dotnet/corefx/blob/148623a7cd0eb2789dfb0d1fd2bfb9fc925baa60/src/System.Text.Json/tests/Utf8JsonReaderTests.cs#L620-L654

Reply:
This test case is for single line comment split across parts (array inputData), each one being treated as a span and processed in iteration. The pattern that you mentioned is cleaner. Will use that for the test case `ConsumeSingleLineCommentSingleSpanTest`. This test case covers the execution paths in Skip/ConsumeSingleLineComment in Utf8JsonReader.cs file.

`SingleLineCommentData` is also used in `ConsumeSingleLineCommentMultiSpanTest`. In this test case, each _part_ of comment i.e. string in `inputData` array itself is broken down into multiple spans to form a `ReadOnlySequence` (ref. [here](https://github.com/dotnet/corefx/pull/33393/files/8bfd86018192bba09b04229545f75bda6c2251a8#diff-297e5e4a465c2b75335c68e4098786d5R562)). The comment is split across multiple `ReadOnlySequence`s which themselves are multi-segment. Idea is to test scenarios in which the line separator may be first character in next segment in same ReadOnlySequence or then first segment of next sequence. This test case covers the code execution paths in Skip/ConsumeSingleLineCommentMultiSegment in Utf8JsonReader.MultiSegment.cs (which has additional logic around `expectLS` and `GetNextSpan`.

Will update the test cases to avoid LINQ usage as far as possible.

Reply:
> `SingleLineCommentData` is also used in `ConsumeSingleLineCommentMultiSpanTest`. In this test case, each _part_ of comment i.e. string in `inputData` array itself is broken down into multiple spans to form a `ReadOnlySequence`

If I understand your `ReadOnlySequence` test scenario correctly, we already have this type of test pattern covered, for example here: https://github.com/dotnet/corefx/blob/a6f76f4f620cbe74821c6445af3f13e048361658/src/System.Text.Json/tests/Utf8JsonReaderTests.MultiSegment.cs#L132-L169

Consider following that approach for these new tests too (even for the multi-segment case).

`JsonTestHelper.GetSequences` splits the sequence at all indices, and for each split sequence, we enumerate and pass a portion of the sequence to the reader.

Given that your inputs are small, I think the test doesn't need to be outerloop, but please verify how long it would take to run such a test (with nested loops). If it adds more than a few seconds to the total test time, we should move to outerloop.

Reply:
hmm. I adopted the pattern by making relevant changes but it is giving me a list of exceptions. Looks like it will take time to figure out how to make sequences adjust to the test case. Will check that tomorrow. Alternatively, isn't the structure that I have written acceptable. I will do all the modifications that you have mentioned below.

Reply:
> Will check that tomorrow. Alternatively, isn't the structure that I have written acceptable. 

It was mainly a nit (and it's tests). Don't block on it :)

The structure you have is fine (modulo the recent review comments). Once those are resolved, we should be able to merge it!

Reply:
Ok thanks :-) Just pushed a commit which gets rid of the Verify Info...